options.go

```diff
@@ -79,3 +79,10 @@ func WithAggregation(a bool) Option {
 		c.isAggregated = a
 	}
 }
+
+//WithBatchSecondInterval overrides the batch retrieval interval for the consumer
+func WithBatchSecondInterval(k int64) Option {
+	return func(c *Consumer) {
```

Comment:
Is this supposed to take a pointer? Usually `With` methods don't usually mutate the object.

Reply:
I believe it is, I followed the same schema of the other With functions in this file

Reply:
😬